connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/GcsVendedTokenProvider.java

```diff
@@ -0,0 +1,36 @@
+package io.unitycatalog.connectors.spark;
+
+import com.google.cloud.hadoop.util.AccessTokenProvider;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.IOException;
+import java.time.Instant;
+
+public class GcsVendedTokenProvider implements AccessTokenProvider {
```

Comment:
We created this class only for using the two static constants?

Reply:
Yeah, this was the only way I was able to get those creds flowing through (if there's a simpler way, LMK!).  Based on the [gcs-connector](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcs/CONFIGURATION.md#user-credentials) docs, authentication can be provided by a few mechanisms:

```
fs.gs.auth.type (default: COMPUTE_ENGINE)

What type of authentication mechanism to use for Google Cloud Storage access.

Valid values:

    ACCESS_TOKEN_PROVIDER - configures AccessTokenProvider authentication
    APPLICATION_DEFAULT - configures [Application Default Credentials](https://javadoc.io/doc/com.google.auth/google-auth-library-oauth2-http/latest/com/google/auth/oauth2/GoogleCredentials.html) authentication
    COMPUTE_ENGINE - configures Google Compute Engine service account authentication
    SERVICE_ACCOUNT_JSON_KEYFILE - configures JSON keyfile service account authentication
    UNAUTHENTICATED - configures unauthenticated access
    USER_CREDENTIALS - configure [user credentials](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcs/CONFIGURATION.md#user-credentials)
```

`ACCESS_TOKEN_PROVIDER` is the only one we can really customize -- all the other options look like they expect either json keyfile credentials, metadata server, or an id/secret pair for initiating an oauth2 flow.  

That requires a class that implements the `AccessTokenProvider` interface -- this is a simple implementation of that interface that just grabs the required values from conf.  Those values should get set after the `generateTemporaryTableCredentials` call.